tests/h/search/query_test.py

```diff
@@ -999,16 +992,15 @@ def test_it_limits_number_of_annotation_counts_by_user_returned(
         result = search.run(webob.multidict.MultiDict({}))
 
         users_results = result.aggregations["users"]
-        count_pb = next(r for r in users_results if r["user"] == "acct:pb@example.com")[
-            "count"
-        ]
-        count_pc = next(r for r in users_results if r["user"] == "acct:pc@example.com")[
-            "count"
-        ]
-
         assert len(users_results) == bucket_limit
-        assert count_pb == 3
-        assert count_pc == 2
+
+        by_user = group_by_key(users_results, "user")
+        assert by_user["acct:pb@example.com"]["count"] == 3
+        assert by_user["acct:pc@example.com"]["count"] == 2
+
+
+def group_by_key(dict_items, key):
+    return {dict_item[key]: dict_item for dict_item in dict_items}
```

Comment:
I don't know for sure that you can't have the same key come back twice, but it seems like you can't in the examples given. As they are grouped by a particular thing, like tag or user, it would be weird if it did.